draft-yorgos-dnsop-dry-run-dnssec.md

```diff
@@ -0,0 +1,302 @@
+      answers for this one. It guarantees that the DNSSEC of the zone is not
+      altered.
+    * If going back to real DS, the real DS is now cached and no EDER reports
+      for the same dry-run DS should be generated.
+
+## wet-run clients {#wet-run-clients}
+
+Wet-run clients are clients that send the EDNS0 option code TBD (Wet-Run
+DNSSEC) when querying a validating resolver. These clients opt-in to receive
+error responses in case of DNSSEC errors in a dry-run zone. They allow for
+end-to-end DNSSEC testing in a controlled environment.
+
+Validating resolvers that recognise the option MUST respond with the error that
+they would normally respond for a DNSSEC zone and MUST attach the same EDNS0
+option code TBD in the response to mark the error response as coming from a
+dry-run zone.
+
+Additional Extended DNS Errors can also be attached in the error response by
+the validating resolver as per [@!RFC8914].
+
+
+# Implementation Notes {#implementation-notes}
+
+TODO
+tldr; validating resolvers need to keep an additional DNSSEC status for cached
+records that notes the DNSSEC status for the dry-run part. Responses can then
+be provided based on the Wet-Run DNSSEC EDNS0 option.
+
+
+# Security Considerations {#security}
+
+Dry-run DNSSEC disables one of the fundamental guarantees of DNSSEC, data
+integrity. Bogus answers for expired/invalid data will become insecure
+answers providing the potentially wrong information back to the requester. This
+is a feature of this proposal but it also allows forged answers by third
+parties to still affect the zone. This should be treated as a warning that
+dry-run DNSSEC is not an end solution but rather a temporarily intermediate
+test step of a zone going secure.
+
+Parent zones that provide signed delegations to child zones should be aware
+that by using dry-run DNSSEC (e.g., testing a key roll to a stronger algorithm
+key) they risk the DNSSEC status of the child zones. If the trust chain becomes
+invalid between parent and child because of dry-run DNSSEC the child zone will
+be treated as insecure.
+
+
+# IANA Considerations
+
+## A New Wet-Run EDNS0 Option
+
+This document updates the IANA registry for EDNS0 options, currently called
+"DNS EDNS0 Option Codes (OPT)". The following entry has been added:
+
+Value          TBD
+Name           Wet-Run DNSSEC
+Status         Standard
+Reference      This document
+
+## A New DRY-RUN DS Type Digest Algorithm
+
+This document updates the IANA registry for digest types in DS records,
+currently called "Delegation Signer (DS) Resource Record (RR) Type Digest
+Algorithms".  The following entry has been added:
+
+Value          TBD
+Digest Type    DRY-RUN
+Status         OPTIONAL (MANDATORY?)
+
+
+# Acknowledgements
+
+This document is based on an idea by Yorgos Thessalonikefs and Willem Toorop.
+Roy Arends contributed the structure of the dry-run DS for the digest type and
+the digest content containing the intended digest type.
```

Comment:
We don't need to Acknowledge ourselves :)

```suggestion
```

Reply:
Maybe :) I was looking at the acknowledgment section of DNSSEC Error Reporting